src/compiler/types.ts

```diff
@@ -1525,7 +1526,9 @@ module ts {
     export interface SymbolLinks {
         target?: Symbol;                    // Resolved (non-alias) target of an alias
         type?: Type;                        // Type of value symbol
-        declaredType?: Type;                // Type of class, interface, enum, or type parameter
+        declaredType?: Type;                // Type of class, interface, enum, type alias, or type parameter
+        typeParameters?: TypeParameter[];   // Type parameters of type alias (empty array if none)
+        instantiations?: Map<Type>;         // Instantiations of generic type alias

```

Comment:
I will think about it. It is generally more convenient to have them on the Type object because that's what we're operating on in most cases. Getting back to the SymbolLinks from the type requires us to do `getSymbolLinks(type.symbol)` which adds a bit of overhead.